classes/UpgradeTools/CoreUpgrader/CoreUpgrader.php

```diff
@@ -689,6 +701,75 @@ protected function updateTheme()
         }
     }
 
+    protected function updateRTLFiles()
+    {
+        if (!$this->container->getUpgradeConfiguration()->shouldUpdateRTLFiles()) {
+            return;
+        }
+
+        // BO theme
+        if (class_exists(RtlStylesheetProcessor::class)) {
+            $this->logger->info($this->container->getTranslator()->trans('Upgrade the RTL files of back-office themes.', [], 'Modules.Autoupgrade.Admin'));
+
+            $this->removeExistingRTLFiles([
+                ['directory' => $this->container->getProperty(UpgradeContainer::PS_ADMIN_PATH) . DIRECTORY_SEPARATOR . 'themes'],
+            ]);
+
+            (new RtlStylesheetProcessor(
+                $this->container->getProperty(UpgradeContainer::PS_ADMIN_PATH),
+                $this->container->getProperty(UpgradeContainer::PS_ROOT_PATH) . DIRECTORY_SEPARATOR . 'themes',
+                []
+            ))
+                ->setProcessBOTheme(true)
+                ->setProcessDefaultModules(true)
+                ->process();
+        }
+
+        // FO themes
+        if (!class_exists(AdaptThemeToRTLLanguagesCommand::class)) {
+            return;
+        }
+
+        $this->logger->info($this->container->getTranslator()->trans('Upgrade the RTL files of front-office themes.', [], 'Modules.Autoupgrade.Admin'));
+        $themeAdapter = new ThemeAdapter($this->db);
+
+        $themes = $themeAdapter->getListFromDisk();
+        $this->removeExistingRTLFiles($themes);
+
+        foreach ($themes as $theme) {
+            $adaptThemeToTRLLanguages = new AdaptThemeToRTLLanguagesCommand(
+                new ThemeName($theme['name'])
+            );
+
+            /** @var CommandBusInterface $commandBus */
+            $commandBus = $this->container->getModuleAdapter()->getCommandBus();
+
+            try {
+                $commandBus->handle($adaptThemeToTRLLanguages);
+            } catch (CoreException $e) {
+                $this->logger->error('
+                    [ERROR] PHP Impossible to generate RTL files for theme' . $theme['name'] . "\n" .
+                    $e->getMessage()
+                );
+
+                $this->container->getState()->setWarningExists(true);
+            }
+        }
+    }
+
+    /**
+     * @param array{array{'directory':string}} $themes
+     */
+    private function removeExistingRTLFiles(array $themes)
+    {
+        $filesystem = new Filesystem();
+
+        foreach ($themes as $theme) {
+            $files = $this->container->getFilesystemAdapter()->listSampleFiles($theme['directory'], '_rtl.css');
+            $filesystem->remove($files);
+        }
```

Comment:
Since the function is called several times, we could instantiate FileSystem earlier to avoid doing it several times